### DISCUSSION TITLE
6x6 Volvo Articulated Hauler - Xbox Controller

### DISCUSSION THREAD

**Discussion by bbaraban** [pybricks/support#1466] [Show and tell]
I adapted the existing [Hauler remote control](https://pybricks.com/projects/sets/technic/42114-volvo-articulated-hauler/powered-up-remote/) (credit: [Repkovsky](https://github.com/repkovsky). firmware to work with the Xbox controller. Steering is now fluid instead of just hard left or right.

Left Trigger - Drive power forward
Right Trigger - Drive power backwards
Left joystick - Steering
Y - Changes Mode from Auto/Manual

Auto Mode
A - Drive Mode
X - Dumper (Left Trigger raise, Right Trigger dump)

Manual Mode
A - Shift down
X - Shift up (gear 4 is dumper)

For instructions on how install check out the Audi project for uploading the firmware and pairing the controller. (hint, have the controller in pair mode when you first upload the program to the Hauler)

https://gist.github.com/bbaraban/71b31c0cc92fab77b9bd850072f6fd44

**Comment by gibona** [comment]
I modified your code and added some improvements:
https://github.com/gibona/pybricks/blob/master/Lego%2042114%206x6%20Volvo%20Articulated%20Hauler%20-%20PyBricks%20-%20Xbox%20Controller.py

+ Fix the Xbox buttons
+ Drive Dumper via RB and LB buttons
+ Added rumble feedback for the selected gear
